lace/emulator/nn_emulator.py

```diff
@@ -108,435 +113,121 @@ def __init__(
-
-        elif (training_set is not None) & (archive is None):
-            if training_set not in training_set_all:
+        self.device = torch.device(
+            "cuda" if torch.cuda.is_available() else "cpu"
+        )
+        training_set_all = list(TrainingSet)
+        emulator_label_all = list(EmulatorLabel)
+        self.GADGET_LABELS = GADGET_LABELS
+        self.NYX_LABELS = NYX_LABELS
+
+        if isinstance(emulator_label, str):
+            try:
+                self.emulator_label = EmulatorLabel(emulator_label)
+            except ValueError:
+                valid_labels = ", ".join(label.value for label in EmulatorLabel)
                 raise ValueError(
-                    f"Invalid training_set value {training_set}. Available options:",
-                    training_set_all,
-                )
-
-            self.print(f"Selected training set {training_set}")
-
-            if training_set in ["Pedersen21", "Cabayol23"]:
-                archive = gadget_archive.GadgetArchive(postproc=training_set)
-            elif training_set[:5] in ["Nyx23"]:
-                archive = nyx_archive.NyxArchive(
-                    nyx_version=training_set[6:], nyx_file=nyx_file
+                    f"Invalid emulator_label: '{emulator_label}'. "
+                    f"Available options are: {valid_labels}"
                 )
 
-            self.training_data = archive.get_training_data(
-                emu_params=self.emu_params,
-                drop_sim=self.drop_sim,
-                z_max=self.z_max,
-            )
-
-        elif (training_set is None) & (archive is not None):
+        if emulator_label in EMULATOR_PARAMS:
             self.print(
-                "Use custom archive provided by the user to train emulator"
-            )
-            self.training_data = archive.get_training_data(
-                emu_params=self.emu_params,
-                drop_sim=self.drop_sim,
-                drop_z=self.drop_z,
-                z_max=self.z_max,
+                EMULATOR_DESCRIPTIONS.get(
+                    emulator_label, "No description available."
+                )
             )
 
-        elif (training_set is not None) & (archive is not None):
-            if train:
-                raise ValueError(
-                    "Provide either archive or training set for training"
-                )
-            else:
-                self.print(
-                    "Using custom archive provided by the user to load emulator"
-                )
-                self.training_data = archive.get_training_data(
-                    emu_params=self.emu_params,
-                    drop_sim=self.drop_sim,
-                    drop_z=self.drop_z,
-                    z_max=self.z_max,
-                )
+            params = EMULATOR_PARAMS[emulator_label]
+            for key, value in params.items():
+                setattr(self, key, value)
+
+    
+        self.archive, self.training_data = select_training(
```

Comment:
Having an archive in each emulator is too much, the memory footprint gets huge when using MPI. Do we really need it? We did not have it in the previous version. Same goes for training_data, do we really need the full training_data when reading an emulator, or it would be enough storing a few things? The main limitation of cup1d right now is the memory footprint